index.ts

```diff
@@ -663,7 +681,7 @@ export const serializeIlpPrepare = (json: IlpPrepare) => {
   writer.write(Buffer.from(dateToInterledgerTime(json.expiresAt), 'ascii'))
   writer.write(json.executionCondition)
   writer.writeVarOctetString(Buffer.from(json.destination, 'ascii'))
-  writer.writeVarOctetString(json.data)
+  writer.writeVarOctetString(json.data || Buffer.alloc(0))
```

Comment:
At some point I think I suggested having default values like this and @justmoon commented that it would be better to make having an empty data thing explicit to avoid unintentional bugs where some variable is accidentally left null or undefined.